app/components/Pagination.tsx

```diff
@@ -0,0 +1,67 @@
+import { ChevronLeftIcon, ChevronRightIcon } from "@chakra-ui/icons";
+import { Box, Flex, HStack } from "@nycplanning/streetscape";
+import { useSearchParams } from "@remix-run/react";
+import { Button } from "@chakra-ui/button";
+import { Link } from "@remix-run/react";
+import { IconButton } from "@chakra-ui/react";
+
+export interface PaginationProps{
+    total: number;
+    path: string,
+};
+
+export const Pagination = ({total, path}: PaginationProps) => {
+    // where do we want to limit to only having 7 as limit
+    const [searchParams, setSearchParams] = useSearchParams();
+    const limit = 7; // Number(searchParams.get("limit"));
+    const offset = Number(searchParams.get("offset"));
+
+    const currentPage = offset === 0 ? 1 : offset / limit + 1;
+    const canSkipBackward = offset > 0;
+    const canSkipForward = total === limit;
+
+    // from https://www.jacobparis.com/content/remix-pagination
+    function setSearchParamsString(
```

Comment:
It seems this is a pure function (it relies solely on the parameters passed into it). If so, we can move its definition to outside the component. This will prevent it being redefined on every component render.

Reply:
Looking at this again, it seems the functionality as a whole is customizes the "useSearchParams" hook to update search parameters. I wonder if it is worth pulling out into a custom hook.

Reply:
Yes exactly. I left it here as is to get it going but tbh, 1. I don't fully understand what's going on in the function (specifically just the `Record` type) and 2. I was wondering if we could customize this function to be a bit more specific to our needs or leave as is.